packages/core/src/types.ts

```diff
@@ -241,4 +322,54 @@ export interface UserOperationStruct {
   /* Data passed into the account along with the nonce during the verification step */
   signature: BytesLike;
 }
+//#endregion UserOperationStruct_v6
+
+//#region UserOperationStruct_v7
+// based on https://github.com/eth-infinitism/account-abstraction/blob/releases/v0.7/test/UserOperation.ts
+// this is used for building requests for v0.7 entry point contract
+export interface UserOperationStruct_v7 {
+  /* the account making the operation */
+  sender: string;
+  /* anti-replay parameter. nonce of the transaction, returned from the entry point for this address */
+  nonce: BigNumberish;
+  /* account factory, only for new accounts */
+  factory: string | "0x";
+  /* data for account factory (only if account factory exists) */
+  factoryData: BytesLike | "0x";
+  /* the data to pass to the sender during the main execution call */
+  callData: BytesLike;
+  /* the amount of gas to allocate the main execution call */
+  callGasLimit?: BigNumberish;
+  /* the amount of gas to allocate for the verification step */
+  verificationGasLimit?: BigNumberish;
+  /* extra gas to pay the bunder */
+  preVerificationGas?: BigNumberish;
+  /* maximum fee per gas (similar to EIP-1559 max_fee_per_gas) */
+  maxFeePerGas?: BigNumberish;
+  /* maximum priority fee per gas (similar to EIP-1559 max_priority_fee_per_gas) */
+  maxPriorityFeePerGas?: BigNumberish;
+  /* address of paymaster contract, (or empty, if account pays for itself) */
+  paymaster: string | "0x";
+  /* the amount of gas to allocate for the paymaster validation code */
+  paymasterVerificationGasLimit?: BigNumberish;
+  /* the amount of gas to allocate for the paymaster post-operation code */
+  paymasterPostOpGasLimit?: BigNumberish;
+  /* data for paymaster (only if paymaster exists) */
+  paymasterData: BytesLike | "0x";
+  /* data passed into the account to verify authorization */
+  signature: BytesLike;
+}
+//#endregion UserOperationStruct_v7
+
+//#region UserOperationStruct
+export type UserOperationStruct<TEntryPointVersion extends EntryPointVersion> =
+  TEntryPointVersion extends "0.6.0"
+    ? UserOperationStruct_v6
+    : TEntryPointVersion extends "0.7.0"
+    ? UserOperationStruct_v7
+    : never;
 //#endregion UserOperationStruct
+
+export type UserOperationLike<
```

Comment:
this type is kinda confusing because it's checking only against UserOperationOverrides, I expected it to be more like UserOperationStruct or Request.

How it used and why is it used over the overrides?

Reply:
using overrides should work technically, but I want to check if all fields exist, even if undefined value fields.

Reply:
what I'm saying it only works with Overrides, but not the other UserOperation type structs. I think we should update the name here

Reply:
do not really need this anymore now. so this is removed. Think this causes more confusion, so better to be clearer.